### PR TITLE
Fix 0 default value for repo snapshot speed

### LINKED ISSUE #93651: [CI] amazon-2&&immutable failing multiple times
### CI Link

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+8.6+multijob+platform-support-unix/os=amazon-2&&immutable/170/

### Repro line

./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.action.admin.ReloadSecureSettingsIT.testInvalidKeystoreFile" -Dtests.seed=E93B0D7895404FAB -Dtests.locale=ms -Dtests.timezone=Asia/Dubai -Druntime.java=17

### Does it reproduce?

Didn't try

### Applicable branches

main, 8.6, 8.7, 7.17

### Failure history

See below

### Failure excerpt

Failure histories:

* 8.6 https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+8.6+multijob+platform-support-unix/os=amazon-2&&immutable/
* 8.7 https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+8.7+multijob+platform-support-unix/os=amazon-2&&immutable/
* 7.17 https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.17+multijob+platform-support-unix/os=amazon-2&&immutable/
* main https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+main+multijob+platform-support-unix/os=amazon-2&&immutable/

Full console is almost 800 MiB! Beginning shows:

```
<span class="timestamp"><b>09:54:33</b> </span>
<span class="timestamp"><b>09:54:33</b> </span>REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.action.admin.ReloadSecureSettingsIT.testInvalidKeystoreFile" -Dtests.seed=E93B0D7895404FAB -Dtests.locale=ms -Dtests.timezone=Asia/Dubai -Druntime.java=17
<span class="timestamp"><b>09:54:33</b> </span>
<span class="timestamp"><b>09:54:33</b> </span>org.elasticsearch.action.admin.ReloadSecureSettingsIT &gt; testInvalidKeystoreFile FAILED
<span class="timestamp"><b>09:54:33</b> </span>    java.lang.AssertionError
<span class="timestamp"><b>09:54:33</b> </span>        at org.junit.Assert.fail(Assert.java:86)
<span class="timestamp"><b>09:54:33</b> </span>        at org.junit.Assert.assertTrue(Assert.java:41)
<span class="timestamp"><b>09:54:33</b> </span>        at org.junit.Assert.assertFalse(Assert.java:64)
<span class="timestamp"><b>09:54:33</b> </span>        at org.junit.Assert.assertFalse(Assert.java:74)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.InternalTestCluster.validateClusterFormed(InternalTestCluster.java:1215)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.InternalTestCluster.startAndPublishNodesAndClients(InternalTestCluster.java:1746)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.InternalTestCluster.reset(InternalTestCluster.java:1192)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.InternalTestCluster.beforeTest(InternalTestCluster.java:1085)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.ESIntegTestCase.lambda$beforeInternal$0(ESIntegTestCase.java:362)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedContext.runWithPrivateRandomness(RandomizedContext.java:187)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedContext.runWithPrivateRandomness(RandomizedContext.java:211)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.ESIntegTestCase.beforeInternal(ESIntegTestCase.java:371)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.ESIntegTestCase.setupTestCluster(ESIntegTestCase.java:2228)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:980)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/java.lang.Thread.run(Thread.java:833)

<span class="timestamp"><b>09:54:33</b> </span>    java.lang.AssertionError: ClusterHealthResponse has timed out - returned: [{"cluster_name":"SUITE-TEST_WORKER_VM=[148]-CLUSTER_SEED=[-6642037413227423824]-HASH=[1047FDDA10D]-cluster","status":"green","timed_out":true,"number_of_nodes":5,"number_of_data_nodes":2,"active_primary_shards":0,"active_shards":0,"relocating_shards":0,"initializing_shards":0,"unassigned_shards":0,"delayed_unassigned_shards":0,"number_of_pending_tasks":0,"number_of_in_flight_fetch":0,"task_max_waiting_in_queue_millis":0,"active_shards_percent_as_number":100.0}]
<span class="timestamp"><b>09:54:33</b> </span>    Expected: is &lt;false&gt;
<span class="timestamp"><b>09:54:33</b> </span>         but: was &lt;true&gt;
<span class="timestamp"><b>09:54:33</b> </span>        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertNoTimeout(ElasticsearchAssertions.java:99)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.ESIntegTestCase.ensureClusterSizeConsistency(ESIntegTestCase.java:1088)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:574)
<span class="timestamp"><b>09:54:33</b> </span>        at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2244)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
<span class="timestamp"><b>09:54:33</b> </span>        at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
<span class="timestamp"><b>09:54:33</b> </span>        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
<span class="timestamp"><b>09:54:33</b> </span>        at java.base/java.lang.Thread.run(Thread.java:833)
<span class="timestamp"><b>09:54:39</b> </span>
<span class="timestamp"><b>09:54:39</b> </span>REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.action.admin.cluster.repositories.RepositoryBlocksIT.testDeleteRepositoryWithBlocks" -Dtests.seed=E93B0D7895404FAB -Dtests.locale=es-US -Dtests.timezone=SystemV/MST7 -Druntime.java=17
<span class="timestamp"><b>09:54:39</b> </span>
```

Then comes a very long list of similar errors like:

```
<span class="timestamp"><b>09:54:43</b> </span>  1&gt; [2023-02-09T00:53:08,818][ERROR][o.e.t.TcpTransport       ] [node_t0] exception from server channel caught on transport layer [Netty4TcpChannel{localAddress=/127.0.0.1:17711}]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt; java.io.IOException: Invalid argument
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at sun.nio.ch.Net.accept(Native Method) ~[?:?]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at sun.nio.ch.ServerSocketChannelImpl.implAccept(ServerSocketChannelImpl.java:425) ~[?:?]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:391) ~[?:?]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.util.internal.SocketUtils$5.run(SocketUtils.java:119) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.util.internal.SocketUtils$5.run(SocketUtils.java:116) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at java.security.AccessController.doPrivileged(AccessController.java:569) ~[?:?]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.util.internal.SocketUtils.accept(SocketUtils.java:116) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at org.elasticsearch.transport.netty4.CopyBytesServerSocketChannel.doReadMessages(CopyBytesServerSocketChannel.java:45) ~[transport-netty4-8.6.2-SNAPSHOT.jar:8.6.2-SNAPSHOT]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:79) ~[netty-transport-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) ~[netty-transport-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:689) ~[netty-transport-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:652) ~[netty-transport-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) ~[netty-transport-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.84.Final.jar:4.1.84.Final]
<span class="timestamp"><b>09:54:43</b> </span>  1&gt;  at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### LINKED ISSUE #95561: Rate limit for snapshot speed is not tied to the recovery rate limit computed from `node.bandwidth.recovery*` settings
### Elasticsearch Version

8.6.2

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Elasticsearch Service (ESS)

### Problem Description

Rate limit for snapshot speed is not tied to the recovery rate limit computed from `node.bandwidth.recovery*` settings.

This happens irregardless of the data tiers where the data is located. 

This is likely a bug introduced in https://github.com/elastic/elasticsearch/pull/91021 . **Issue has been observed by users and it is reproducible.**

Tagging @tlrx @kingherc as we have internally discussed about this.

### Steps to Reproduce

1) Create a deployment in Elasticsearch Service (ESS):
- Hardware profile: Storage optimized (dense)
- Hot tier: 60GB RAM x 1 availability zone (AZ)
- Warm tier: 60GB RAM x 1 availability zone (AZ)

Once the deployment is created, check the `node.bandwidth.recovery*` settings and check the Elasticsearch logs for the computed rate limit.

From the [nodes info API](https://www.elastic.co/guide/en/elasticsearch/reference/8.6/cluster-nodes-info.html) output, extract the `node.bandwidth.recovery` setting using `jq`:

```
jq '.nodes[]|{name:.name, instance_configuration:.attributes.instance_configuration, bandwidth:.settings.node.bandwidth}' nodes.json
```

```
{
  "name": "instance-0000000000",
  "instance_configuration": "aws.es.datahot.i3en",
  "bandwidth": {
    "recovery": {
      "disk": {
        "write": "636200Kb",
        "read": "1249280Kb"
      },
      "network": "1018060Kb"
    }
  }
}
{
  "name": "instance-0000000001",
  "instance_configuration": "aws.es.datawarm.d3",
  "bandwidth": {
    "recovery": {
      "disk": {
        "write": "954255Kb",
        "read": "293795Kb"
      },
      "network": "727541Kb"
    }
  }
}
```

In the Elasticsearch logs, we can find the computed rate limit for the two nodes:

```
[2023-04-26T00:14:00,061][INFO ][org.elasticsearch.indices.recovery.RecoverySettings] [instance-0000000000] using rate limit [248.5mb] with [default=40mb, read=397.6mb, write=248.5mb, max=60.6gb]
[2023-04-26T00:14:00,866][INFO ][org.elasticsearch.indices.recovery.RecoverySettings] [instance-0000000001] using rate limit [114.7mb] with [default=40mb, read=114.7mb, write=284.1mb, max=28gb]
```

2) Stop SLM (`POST _slm/stop`)

3) Using [rally](https://github.com/elastic/rally), run the `nyc_taxis` track with the `append-no-conflicts-index-only` challenge to create the `nyc_taxis` index, which should be about 20-25GB in size.

4) Execute the SLM policy to take a snapshot:

```
PUT _slm/policy/cloud-snapshot-policy/_execute
```

5) Wait for the snapshot to complete:

```
[2023-04-26T01:13:46,029][INFO ][org.elasticsearch.xpack.slm.SnapshotLifecycleTask] [instance-0000000000] snapshot lifecycle policy [cloud-snapshot-policy] issuing create snapshot [cloud-snapshot-2023.04.26-4olr-h_lq_63ljveqiwr2q]
[2023-04-26T01:13:46,055][INFO ][org.elasticsearch.snapshots.SnapshotsService] [instance-0000000000] snapshot [found-snapshots:cloud-snapshot-2023.04.26-4olr-h_lq_63ljveqiwr2q/VLurh60aT_qeeSlsEGcWZg] started
[2023-04-26T01:24:14,821][INFO ][org.elasticsearch.snapshots.SnapshotsService] [instance-0000000000] snapshot [found-snapshots:cloud-snapshot-2023.04.26-4olr-h_lq_63ljveqiwr2q/VLurh60aT_qeeSlsEGcWZg] completed with state [SUCCESS]
```

Observe the host network inbound/outbound utilisation. While the snapshot was taken, we can see that the snapshot speed was limited to ~ 40mb/s:

snapshot-in-hot-tier-default-settings

**This is not the expected behaviour.**

6) Delete the snapshot

7) Update the snapshot repository `found-snapshots` with `max_snapshot_bytes_per_sec` to `248mb` (to roughly match the expected rate limit in the hot data tier)

8) Execute the SLM policy to take a snapshot

9) Wait for the snapshot to complete:

```
[2023-04-26T01:46:44,787][INFO ][org.elasticsearch.xpack.slm.SnapshotLifecycleTask] [instance-0000000000] snapshot lifecycle policy [cloud-snapshot-policy] issuing create snapshot [cloud-snapshot-2023.04.26-mwgisfissvq1uvorxigoyg]
[2023-04-26T01:46:44,797][INFO ][org.elasticsearch.snapshots.SnapshotsService] [instance-0000000000] snapshot [found-snapshots:cloud-snapshot-2023.04.26-mwgisfissvq1uvorxigoyg/3rjPoMI-Ti6pXJStt24GNg] started
[2023-04-26T01:48:28,709][INFO ][org.elasticsearch.snapshots.SnapshotsService] [instance-0000000000] snapshot [found-snapshots:cloud-snapshot-2023.04.26-mwgisfissvq1uvorxigoyg/3rjPoMI-Ti6pXJStt24GNg] completed with state [SUCCESS]
```

Observe the host network inbound/outbound utilisation. While the snapshot was taken, we can see that the snapshot speed was up to ~ 250mb/s:

snapshot-in-hot-tier-with-custom-snapshot-speed

10) Delete the snapshot and remove the `max_snapshot_bytes_per_sec` setting from the snapshot repository configuration

11) Update the `_tier_preference` for `nyc_taxis` index so that the index moves to the warm data tier:

```
PUT nyc_taxis/_settings
{
  "index": {
    "routing": {
      "allocation": {
        "include": {
          "_tier_preference": "data_warm,data_hot"
        }
      }
    }
  }
}
```

12) Wait for the primary shard to be moved to the warm data tier

13) Execute the SLM policy to take a snapshot

14) Wait for the snapshot to complete:

```
[2023-04-26T02:19:25,635][INFO ][org.elasticsearch.xpack.slm.SnapshotLifecycleTask] [instance-0000000000] snapshot lifecycle policy [cloud-snapshot-policy] issuing create snapshot [cloud-snapshot-2023.04.26-8qo6hzwksmqi_6dnd7m_hq]
[2023-04-26T02:19:25,650][INFO ][org.elasticsearch.snapshots.SnapshotsService] [instance-0000000000] snapshot [found-snapshots:cloud-snapshot-2023.04.26-8qo6hzwksmqi_6dnd7m_hq/kioCVM4YSpii7lK3iRVC6g] started
[2023-04-26T02:29:53,481][INFO ][org.elasticsearch.snapshots.SnapshotsService] [instance-0000000000] snapshot [found-snapshots:cloud-snapshot-2023.04.26-8qo6hzwksmqi_6dnd7m_hq/kioCVM4YSpii7lK3iRVC6g] completed with state [SUCCESS]
```

Observe the host network inbound/outbound utilisation. While the snapshot was taken, we can see that the snapshot speed was limited to ~ 40mb/s:

snapshot-in-warm-tier-default-settings

**This is not the expected behaviour**

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by kingherc**
When node bandwidth recovery settings are defined.

Fixes #95561

### CODE REVIEW

**Comment by kingherc** [comment]
I have also manually tested this with the methodology mentioned on the related bug ticket, albeit on a small dataset, and it works. I will test on the larger dataset that the ticket uses tomorrow to confirm, but I expect the same result. Thus this is ready for review.

**Comment by kingherc** [comment]
@elasticmachine please run elasticsearch-ci/part-1

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -1660,17 +1653,29 @@ private RateLimiter getRateLimiter(
         }
     }
 
-    private RateLimiter getSnapshotRateLimiter() {
+    // package private for testing
+    RateLimiter getSnapshotRateLimiter() {
+        Settings repositorySettings = metadata.settings();
+        logger.warn("Repository settings: [{}]", repositorySettings);
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:1659]: Do not forget to remove this :)

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 2]
```diff
@@ -1660,17 +1653,29 @@ private RateLimiter getRateLimiter(
...
+        ByteSizeValue maxConfiguredBytesPerSec = MAX_SNAPSHOT_BYTES_PER_SEC.get(repositorySettings);
+        if (MAX_SNAPSHOT_BYTES_PER_SEC.exists(repositorySettings) == false && recoverySettings.nodeBandwidthSettingsExist()) {
+            assert maxConfiguredBytesPerSec.getMb() == 40;
+            maxConfiguredBytesPerSec = ByteSizeValue.ZERO;
+            logger.warn("Setting default to: [{}] because of node bandwidth recovery settings", maxConfiguredBytesPerSec);
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:1664]: And this too

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 3]
```diff
@@ -1660,17 +1653,29 @@ private RateLimiter getRateLimiter(
...
         );
     }
 
-    private RateLimiter getRestoreRateLimiter() {
-        return getRateLimiter(restoreRateLimiter, metadata.settings(), MAX_RESTORE_BYTES_PER_SEC, true);
+    // package private for testing
+    RateLimiter getRestoreRateLimiter() {
+        Settings repositorySettings = metadata.settings();
+        ByteSizeValue maxConfiguredBytesPerSec = MAX_RESTORE_BYTES_PER_SEC.get(repositorySettings);
+        return getRateLimiter(restoreRateLimiter, maxConfiguredBytesPerSec, MAX_RESTORE_BYTES_PER_SEC.getKey(), true);
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:1678]:
```suggestion
        return getRateLimiter(restoreRateLimiter, MAX_RESTORE_BYTES_PER_SEC.get(metadata.settings()), MAX_RESTORE_BYTES_PER_SEC.getKey(), true);
```

**Review by tlrx — APPROVED** [review]
LGTM - Please do not forget to remove some logs and update the description with the correct issue number

**Comment by kingherc** [comment]
Thanks @tlrx ! I am verifying manually with a cloud build, and then will merge the PR.

**Pull request merged**